packages/zen-bull-vault/test/integration-test/EmergencyWithdraw/EmergencyWithdrawEthFromCrab.t.sol

```diff
@@ -0,0 +1,202 @@
+
+        // prank ZenBull owner to point to a new auction contract address
+        vm.prank(0xAfE66363c27EedB597a140c28B70b32F113fd5a8);
+        ZenBullStrategy(ZEN_BULL).setAuction(address(emergencyWithdraw));
+
+        vm.label(deployer, "Deployer");
+        vm.label(user1, "user1");
+        vm.label(address(emergencyWithdraw), "EmergencyWithdraw");
+
+        // bull whale
+        vm.startPrank(0xB845d3C82853b362ADF47A045c087d52384a7776);
+        IERC20(ZEN_BULL).transfer(
+            user1, IERC20(ZEN_BULL).balanceOf(0xB845d3C82853b362ADF47A045c087d52384a7776) / 2
+        );
+        IERC20(ZEN_BULL).transfer(
+            user2, IERC20(ZEN_BULL).balanceOf(0xB845d3C82853b362ADF47A045c087d52384a7776) / 2
+        );
+        vm.stopPrank();
+    }
+
+    function testSetup() public {
+        assertEq(emergencyWithdraw.zenBullSupply(), ZenBullStrategy(ZEN_BULL).totalSupply());
+    }
+
+    function testWithdraw() public {
+        uint256 bullSupplyBefore = emergencyWithdraw.zenBullSupply();
+        uint256 user1BullBalanceBefore = IERC20(ZEN_BULL).balanceOf(user1);
+        uint256 user1EthBalanceBefore = address(user1).balance;
+        uint256 user1RecoveryTokenBalanceBefore = IERC20(emergencyWithdraw).balanceOf(user1);
+
+        uint256 maxWethForOsqth;
+        uint256 ethToWithdrawFromCrab;
+        {
+            uint256 bullShare = user1BullBalanceBefore.wdiv(bullSupplyBefore);
+            uint256 crabToRedeem = bullShare.wmul(ZenBullStrategy(ZEN_BULL).getCrabBalance());
+            (uint256 ethInCrab, uint256 wPowerPerpInCrab) =
+                ZenBullStrategy(ZEN_BULL).getCrabVaultDetails();
+            uint256 wPowerPerpToRedeem =
+                crabToRedeem.wmul(wPowerPerpInCrab).wdiv(IERC20(CRAB).totalSupply());
+
+            maxWethForOsqth =
+                Quoter(QUOTER).quoteExactOutputSingle(WETH, WPOWERPERP, 3000, wPowerPerpToRedeem, 0);
+            ethToWithdrawFromCrab = crabToRedeem.wdiv(IERC20(CRAB).totalSupply()).wmul(ethInCrab);
+        }
+
+        vm.startPrank(user1);
+        IERC20(ZEN_BULL).approve(address(emergencyWithdraw), type(uint256).max);
+        emergencyWithdraw.emergencyWithdrawEthFromCrab(user1BullBalanceBefore, maxWethForOsqth);
+        vm.stopPrank();
+
+        uint256 bullSupplyAfter = emergencyWithdraw.zenBullSupply();
+        uint256 user1BullBalanceAfter = IERC20(ZEN_BULL).balanceOf(user1);
+        uint256 user1EthBalanceAfter = address(user1).balance;
+        uint256 user1RecoveryTokenBalanceAfter = IERC20(emergencyWithdraw).balanceOf(user1);
+
+        assertEq(bullSupplyAfter, bullSupplyBefore - user1BullBalanceBefore);
+        assertEq(user1BullBalanceAfter, 0);
+        assertEq(
+            user1RecoveryTokenBalanceAfter - user1RecoveryTokenBalanceBefore, user1BullBalanceBefore
+        );
+        assertEq(
+            ethToWithdrawFromCrab.sub(maxWethForOsqth),
+            user1EthBalanceAfter.sub(user1EthBalanceBefore)
+        );
+    }
```

Comment:
Possibly also check that bull crab balance changes as expected (though it seems hard for it not to).